Astronomy/Model/Cache.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  Cache.swift
+//  Astronomy
+//
+//  Created by Cora Jacobson on 9/10/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import Foundation
+
+class Cache<Key: Hashable, Value> {
+    var dictionary: [Key: Value] = [:]
+    
+    func cache(key: Key, value: Value) {
+        queue.async {
+            self.dictionary[key] = value
+        }
+    }
+    
+    subscript(_ key: Key) -> Value? {
+        get {
+            queue.sync {
+                if let value = dictionary[key] {
+                    return value
+                } else {
+                    return nil
+                }
```

Comment:
No need to do an if let, you can simply return dictionary[key] which is an optional Value. In-fact, the whole thing can be simplified to this:

``` swift
subscript(_ key: Key) -> Value? {
    queue.sync { dictionary[key] }
}
```